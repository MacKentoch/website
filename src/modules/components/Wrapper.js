import styled from 'styled-components'

const Wrapper = styled.div`
  display: flex;
  flex-direction: ${props => props.flexDirection};
  max-width: 1442px;
  margin: 70px auto;
  @media (min-width: 700px) {
    margin: 100px auto;
    flex-direction: ${props => props.lgFlexDirection || props.flexDirection};
  }
`

Wrapper.defaultProps = {
  flexDirection: 'row',
}

export default Wrapper
