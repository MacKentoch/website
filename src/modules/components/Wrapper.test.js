import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Wrapper from './Wrapper'

describe('Wrapper', () => {
  it('should render by default', () => {
    const wrapper = shallow(<Wrapper>Hello</Wrapper>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should support "flexDirection"', () => {
    const wrapper = shallow(<Wrapper flexDirection="column">Hello</Wrapper>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should support "lgFlexDirection"', () => {
    const wrapper = shallow(<Wrapper lgFlexDirection="column">Hello</Wrapper>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
