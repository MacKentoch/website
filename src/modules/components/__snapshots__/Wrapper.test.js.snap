// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper should render by default 1`] = `
<div
  className="sc-bdVaJa dfCAjY"
>
  Hello
</div>
`;

exports[`Wrapper should support "flexDirection" 1`] = `
<div
  className="sc-bdVaJa dSHSRf"
>
  Hello
</div>
`;

exports[`Wrapper should support "lgFlexDirection" 1`] = `
<div
  className="sc-bdVaJa dLrJrl"
>
  Hello
</div>
`;
